lib/galaxy_test/selenium/test_history_panel_collections.py

```diff
@@ -132,6 +132,21 @@ def test_output_collection_states_running(self):
             for job in payload["jobs"]:
                 self.dataset_populator.cancel_job(job["id"])
 
+    @selenium_test
+    def test_collection_job_details(self):
+        input_collection, failed_collection = self._generate_partially_failed_collection_with_input()
+        ok_hid = input_collection["hid"]
+        failed_hid = failed_collection["hid"]
+        self.home()
+
+        ok_collection_element = self.history_panel_wait_for_hid_state(ok_hid, "ok")
+        failed_collection_element = self.history_panel_wait_for_hid_state(failed_hid, "error")
+
+        ok_collection_element.collection_job_details_button.wait_for_and_click()
+        self.screenshot("history_panel_collections_job_details_ok")
+        failed_collection_element.collection_job_details_button.wait_for_and_click()
+        self.screenshot("history_panel_collections_job_details_failed")
```

Comment:
I think in both cases you'll want to wait for something specific in the error page, like the tool id for instance ?

Reply:
IOW there's no way for the rendering to have finished before taking the screenshots, but also the screenshots aren't a test in itself, you really have to assert something.

Reply:
Yes, sorry I wasn't clear that these screenshots were taken while I was manually debugging, not just plain running the test as it is right now in the branch (of course those would not render properly). So I could make sure both pages were truly reloaded and also I could inspect the API response in the dev tools.